tests/test_Fstab.py

```diff
@@ -87,4 +93,33 @@ def test_parses_auto_lines(
 		assert str(fstab[0]) == "# This is a comment\twith tabs"
 		assert type(fstab[1]).__name__ == "FstabAutoPlaceholderLine"
 		assert type(fstab[2]).__name__ == "FstabCommentLine"
-		assert str(fstab[2]) == "# yet another comment line"
\ No newline at end of file
+		assert str(fstab[2]) == "# yet another comment line"
+
+	@pytest.fixture(scope="function")
+	def zfs_volume(
+		self,
+		root_dataset: libzfs.ZFSDataset
+	) -> libzfs.ZFSDataset:
+
+		r = random.randint(0, sys.maxint)
+		name = f"{root_dataset.name}/zvol{r}"
+
+		root_dataset.pool.create(
+			dataset_name,
+			fsopts=dict(volsize="16M"),
+			fstype=libzfs.DatasetType.VOLUME
+		)
+
+		dataset = zfs.get_dataset(dataset_name)
+		yield dataset
```

Comment:
should we not wrap this generator with something that calls `newfs`?

Reply:
Would you agree that another fixture `zfs_volume_ufs` could do that job?

Reply:
probably a good idea

Reply:
done